File: BubbleSort.js
process.stdin.resume();
process.stdin.setEncoding('utf8');
// Your code here!

/*===== ランダム =====*/
function getRandomInt(max) {
  return Math.floor(Math.random() * max);
}

/*===== 表示 =====*/
function print(array){
    for(var i = 0; i < array.length; i++){
        console.log("num[" + i + "]=" + array[i]);
    }
}

/*===== バブルソートアルゴリズム =====*/
function sort(array){
    for(var j = 0; j < array.length; j++){
        for(var i = 1; i < array.length; i++){
            if(array[i-1] > array[i]){
                var z = array[i-1];
                array[i-1] = array[i];
                array[i] = z
            }
        }
    }
    return array;
}

/*===== メイン関数 =====*/
function main(){
    let num = [];
    
    for(var i = 0; i < 10; i++){
        num[i] = getRandomInt(100);
    }
    
    console.log("rand");
    print(num);
    num = sort(num);
    console.log("sort");
    print(num);
}

main();

File: BubbleSort.java
import java.util.Random;

class Main{
    /*===== 表示 =====*/
    static void print(int array[]){
        for(int i = 0; i < array.length; i++){
            System.out.println("num[" + i +"] =" + array[i]);
        }
    }
    
    /*===== バブルソートアルゴリズム =====*/
    static void sort(int array[]){
        for(int j = 1; j < array.length; j++){
            for(int i = 1; i < array.length; i++){
                if(array[i-1] >= array[i]){
                    int z = array[i-1];
                    array[i-1] = array[i];
                    array[i] = z;
                }
            }
        }
    }
    
    /*===== メイン =====*/
    public static void main(String[] args){
        int num[] = new int[10];
        boolean flag = true;
        
        Random r = new Random();
        
        for(int i = 0; i < 10; i++){
            num[i] = r.nextInt(100);
        }
        
        System.out.println("rand");
        print(num);
        sort(num);
        System.out.println("sort");
        print(num);
    }
}